src/yunohost/settings.py

```diff
@@ -0,0 +1,189 @@
+import os
+import json
+import errno
+
+from datetime import datetime
+from collections import OrderedDict
+
+from moulinette.core import MoulinetteError
+from moulinette.utils.log import getActionLogger
+
+logger = getActionLogger('yunohost.settings')
+
+SETTINGS_PATH = "/etc/yunohost/settings.json"
+SETTINGS_PATH_OTHER_LOCATION = "/etc/yunohost/settings-%s.json"
+
+# a settings entry is in the form of:
+# name: {type, value, default, description, [possibilities]}
+# possibilities is only for enum
+
+# description is implied from the translated strings
+# the key is "global_settings_setting_%s" % key.replace(".", "_")
+
+# type can be:
+# * bool
+# * int
+# * string
+# * enum (in form a python list)
+
+# we don't store the value in default options
+DEFAULTS = OrderedDict([
+    ("example.bool", {"type": "bool", "default": True}),
+    ("example.int", {"type": "int", "default": 42}),
+    ("example.string", {"type": "string", "default": "yolo swag"}),
+    ("example.enum", {"type": "enum", "default": "a", "choices": ["a", "b", "c"]}),
+])
+
+
+def settings_get(key):
+    settings = _get_settings()
+
+    if key not in settings:
+        raise MoulinetteError(errno.EINVAL, m18n.n(
+            'global_settings_key_doesnt_exists', settings_key=key))
+
+    return settings[key]
+
+
+def settings_list():
+    return _get_settings()
+
+
+def settings_set(key, value):
+    settings = _get_settings()
+
+    if key not in settings:
+        raise MoulinetteError(errno.EINVAL, m18n.n(
+            'global_settings_key_doesnt_exists', settings_key=key))
+
+    key_type = settings[key]["type"]
+
+    if key_type == "bool":
+        if not isinstance(value, bool):
+            raise MoulinetteError(errno.EINVAL, m18n.n(
+                'global_settings_bad_type_for_setting', setting=key,
+                received_type=type(value).__name__, expected_type=key_type))
+    elif key_type == "int":
+        if not isinstance(value, int) or isinstance(value, bool):
+            raise MoulinetteError(errno.EINVAL, m18n.n(
+                'global_settings_bad_type_for_setting', setting=key,
+                received_type=type(value).__name__, expected_type=key_type))
+    elif key_type == "string":
```

Comment:
I don't see this bringing that much value here tbh, I prefer this simplest way because it avoid one indirection (having to look in a dictionary elsewhere thus making the code a little bit harder to read) to the price of not being 100% elegant, but I think that simplicity prevails here.

If those types where serializable in json I would have used them.